# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCRegMgmt-inl.h
# Commit: 3ecda60613c5
# Full Hash: 3ecda60613c5cb7a57cc7004919c7d12c3a29268
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-22 03:08:27
# Description:
#   Bug 1736531 - Fix. r=jseward
#   
#   See bug for analysis.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129010
# ==============================================================================

diff -r 1044b04cf89b -r 3ecda60613c5 js/src/wasm/WasmBCRegMgmt-inl.h
--- a/js/src/wasm/WasmBCRegMgmt-inl.h	Thu Oct 21 15:44:59 2021 +0000
+++ b/js/src/wasm/WasmBCRegMgmt-inl.h	Thu Oct 21 15:46:28 2021 +0000
@@ -365,10 +365,8 @@
   }
 }
 
-#ifdef JS_CODEGEN_X64
-void BaseCompiler::maskResultRegisters(ResultType type) {
-  MOZ_ASSERT(JitOptions.spectreIndexMasking);
-
+#ifdef JS_64BIT
+void BaseCompiler::widenInt32ResultRegisters(ResultType type) {
   if (type.empty()) {
     return;
   }
@@ -376,7 +374,7 @@
   for (ABIResultIter iter(type); !iter.done(); iter.next()) {
     ABIResult result = iter.cur();
     if (result.inRegister() && result.type().kind() == ValType::I32) {
-      masm.movl(result.gpr(), result.gpr());
+      masm.widenInt32(result.gpr());
     }
   }
 }
@@ -451,10 +449,8 @@
 
 void BaseCompiler::captureCallResultRegisters(ResultType type) {
   captureResultRegisters(type);
-#ifdef JS_CODEGEN_X64
-  if (JitOptions.spectreIndexMasking) {
-    maskResultRegisters(type);
-  }
+#ifdef JS_64BIT
+  widenInt32ResultRegisters(type);
 #endif
 }
 